### PR TITLE
DQA-4154: [Toolkit] Ask for credentials when QA_API_BASIC_AUTH  is missing

### PR DESCRIPTION
**Pull request by joaocsilva**

### CODE REVIEW

File: src/TaskRunner/Commands/ToolCommands.php [hunk 1]
```diff
@@ -1203,4 +1210,29 @@ private function getPackagePropertyFromComposer($package, $prop = 'version')
...
+            $this->say('Missing env var QA_API_BASIC_AUTH, asking for access.');
+            if (empty($user = $this->ask('Please insert your username?'))) {
+                $this->writeln('<error>The username cannot be empty!</error>');
+                return '';
+            }
+            if (empty($pass = $this->ask('Please insert your password?', true))) {
+                $this->writeln('<error>The password cannot be empty!</error>');
+                return '';
+            }
+            $auth = base64_encode("$user:$pass");
```
Comment (jonhy81) [inline src/TaskRunner/Commands/ToolCommands.php:1293]: Add feedback with indication to include/store the token in the environment vars?

File: src/TaskRunner/Commands/ToolCommands.php [hunk 2]
```diff
@@ -1203,4 +1210,29 @@ private function getPackagePropertyFromComposer($package, $prop = 'version')
...
+     *
+     * @return string
+     *   The Basic auth or empty string if fails.
+     */
+    public function getQaApiBasicAuth(): string
+    {
+        $auth = getenv('QA_API_BASIC_AUTH');
+        if (empty($auth)) {
+            $this->say('Missing env var QA_API_BASIC_AUTH, asking for access.');
+            if (empty($user = $this->ask('Please insert your username?'))) {
```
Comment (jonhy81) [inline src/TaskRunner/Commands/ToolCommands.php]:
```suggestion
            if (empty($user = $this->ask('Please insert your username:'))) {
```

File: src/TaskRunner/Commands/ToolCommands.php [hunk 3]
```diff
@@ -1203,4 +1210,29 @@ private function getPackagePropertyFromComposer($package, $prop = 'version')
...
+                $this->writeln('<error>The username cannot be empty!</error>');
+                return '';
+            }
+            if (empty($pass = $this->ask('Please insert your password?', true))) {
```
Comment (jonhy81) [inline src/TaskRunner/Commands/ToolCommands.php]:
```suggestion
            if (empty($pass = $this->ask('Please insert your password:', true))) {
```

File: src/TaskRunner/Commands/CloneCommands.php [hunk 4]
```diff
@@ -222,7 +222,7 @@ public function downloadDump(array $options = [
             }
         }
         if (empty($password)) {
-            if (empty($password = $this->ask('Please insert your password?'))) {
+            if (empty($password = $this->ask('Please insert your password?', true))) {
```
Comment (jonhy81) [inline src/TaskRunner/Commands/CloneCommands.php]:
```suggestion
            if (empty($password = $this->ask('Please insert your password:', true))) {
```

**Review by jonhy81 — CHANGES_REQUESTED** [review]

File: src/TaskRunner/Commands/ToolCommands.php (hunk 1 above)
Comment (joaocsilva) [reply src/TaskRunner/Commands/ToolCommands.php:1293]: Added message

**Review by jonhy81 — APPROVED** [review]

**Pull request merged**